plugins/BEdita/Core/src/Model/Table/FoldersTable.php

```diff
@@ -85,6 +85,15 @@ public function buildRules(RulesChecker $rules)
             ]
         );
 
+        $rules->add(
+            [$this, 'isFolderRestorable'],
+            'isFolderRestorable',
+            [
+                'errorField' => 'deleted',
+                'message' => __d('bedita', 'Folder can be restored only if its ancestors are not deleted.'),
+            ]
+        );
```

Comment:
Can this be set as a validation rule?

```php
$validator
    ->add('deleted', 'restorable', [
        'rule' => function ($value, $context) {
            // check that folder's parents are not deleted
        },
        'when' => function ($context) {
            return $context['data']['deleted'] === true;
        },
    ]);
```

Reply:
:thinking: I'm not sure simple validation is the right place to check if a folder is restorable. To me it seems more appropriate as application rule because we are checking the state of the application.

Moreover having already the entity we can check against `deleted` is dirty.

Reply:
Ok then, 👍 for leaving it here.